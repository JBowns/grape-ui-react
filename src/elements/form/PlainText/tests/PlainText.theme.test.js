import React from 'react';
import { PlainText } from '..';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

describe('PlainText using Theme color', () => {
  it('should return a PlainText object with custom color', () => {
    const theme = {
      colors: {
        grapeSoda: {
          base: 'hsl(325, 84.6%, 28%)',
          dark: 'hsl(305, 33.9%, 23.7%)',
          light: 'hsl(313, 67.8%, 47.5%)',
        },
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <PlainText bg="grapeSoda.dark" color="grapeSoda.light">Lorem Ipsum</PlainText>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('PlainText using Theme Font Family', () => {
  it('should return a PlainText object with a custom font family', () => {
    const theme = {
      fonts: {
        trueSpace: '"DejaVu Sans Mono", "Bitstream Vera Sans Mono", Courier',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <PlainText fontFamily="trueSpace">Lorem Ipsum</PlainText>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
