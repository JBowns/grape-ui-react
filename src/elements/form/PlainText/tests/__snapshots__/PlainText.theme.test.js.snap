// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainText using Theme Font Family should return a PlainText object with a custom font family 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: "DejaVu Sans Mono","Bitstream Vera Sans Mono",Courier;
  font-size: inherit;
  font-weight: inherit;
  width: 100%;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  padding-bottom: 1rem;
  padding-top: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div>
    
  </div>
</div>
`;

exports[`PlainText using Theme color should return a PlainText object with custom color 1`] = `
.c0 {
  background-color: hsl(305,33.9%,23.7%);
  color: hsl(313,67.8%,47.5%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  width: 100%;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  padding-bottom: 1rem;
  padding-top: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div>
    
  </div>
</div>
`;
